snuba/query/matchers.py

```diff
@@ -313,7 +313,7 @@ def match(self, node: AnyType) -> Optional[MatchResult]:
                 result = result.merge(partial_result)
 
         if self.parameters:
-            if self.with_optionals is None:
+            if not self.with_optionals:
```

Comment:
well this was dumb, `self.with_optionals` is a boolean. Fortunately no pattern so far relied on this behavior to work properly.